app/services/provider_authorisation.rb

```diff
@@ -59,11 +62,21 @@ def can_make_decisions?(application_choice:, course_option_id:)
   end
 
   def can_view_safeguarding_information?(course:)
-    @actor.provider_permissions.view_safeguarding_information
-      .exists?(provider: [course.provider, course.accredited_provider].compact) &&
-      (course.accredited_provider.blank? ||
-        ratifying_provider_can_view_safeguarding_information?(course: course) ||
-          training_provider_can_view_safeguarding_information?(course: course))
+    errors << :requires_view_safeguarding_information unless
```

Comment:
In two minds about this. We usually check one permission at a time, right? So maybe instead of 

```
requires_view_safeguarding_information
requires_training_provider_can_view_safeguarding_information
requires_ratifying_provider_can_view_safeguarding_information
```

we could get away with

```
requires_provider_user_permission
requires_training_provider_permission
requires_ratifying_provider_permission
```

which is a bit less to chew on for clients? Wdyt?

Reply:
I thought matching the error symbol with the name of the permission is a good idea, but what you've suggested may be better, as it makes error symbol names the same across all different authorisation calls (e.g. `can_view_diversity_information?`). We can easily infer which _specific_ permission is missing because we know we are calling `can_view_safeguarding_information?`, for example.